src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -34,22 +37,31 @@
 {
     const ROUTE_NAME_PREFIX = 'api_';
     const DEFAULT_ACTION_PATTERN = 'api_platform.action.';
+    const SUBRESOURCE_SUFFIX = '_get_subresource';
 
     private $fileLoader;
+    private $propertyNameCollectionFactory;
+    private $propertyMetadataFactory;
     private $resourceNameCollectionFactory;
     private $resourceMetadataFactory;
     private $operationPathResolver;
     private $container;
     private $formats;
 
-    public function __construct(KernelInterface $kernel, ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, OperationPathResolverInterface $operationPathResolver, ContainerInterface $container, array $formats)
+    public function __construct(KernelInterface $kernel, ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, OperationPathResolverInterface $operationPathResolver, ContainerInterface $container, array $formats, PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory = null, PropertyMetadataFactoryInterface $propertyMetadataFactory = null)
     {
         $this->fileLoader = new XmlFileLoader(new FileLocator($kernel->locateResource('@ApiPlatformBundle/Resources/config/routing')));
         $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
         $this->resourceMetadataFactory = $resourceMetadataFactory;
         $this->operationPathResolver = $operationPathResolver;
         $this->container = $container;
         $this->formats = $formats;
+        $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
+        $this->propertyMetadataFactory = $propertyMetadataFactory;
+
+        if (null === $this->propertyNameCollectionFactory || null === $this->propertyMetadataFactory) {
+            @trigger_error('Not injecting "PropertyNameCollectionFactoryInterface" and "PropertyMetadataFactoryInterface" instances in '.__CLASS__.' is deprecrated since API Platform 2.1 and will not be possible anymore in API Platform 3', E_USER_DEPRECATED);
```

Comment:
Is this really necessary? To ease using API Platform as a standalone lib, it would be nice to be able to not provide those instance (and then loose subresource support)